src/main/java/org/truffleruby/language/objects/WarnTopLevelClassVariableAccessNode.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.truffleruby.language.objects;
+
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.source.SourceSection;
+import org.truffleruby.language.RubyBaseNode;
+import org.truffleruby.language.WarnNode;
+
+public class WarnTopLevelClassVariableAccessNode extends RubyBaseNode {
+
+    @Child private WarnNode warnNode = new WarnNode();
+
+    @TruffleBoundary
+    public void warnTopLevelClassVariableAccess(SourceSection sourceSection) {
+        warnNode.warningMessage(sourceSection, "class variable access from toplevel");
```

Comment:
Given the simplicity of this node, I think we could do without now, it would only duplicate the `"class variable access from toplevel"` which seems acceptable. What do you think?

Reply:
Sounds good, fixed.